### PR TITLE
Gra 991 local connection

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

File: models/api_host.go [hunk 1]
```diff
@@ -50,10 +49,7 @@ func (h *Host) ConvertNMHostToAPI() *ApiHost {
 	}
 	a.IsStatic = h.IsStatic
 	a.ListenPort = h.ListenPort
-	a.LocalRange = h.LocalRange.String()
-	if isEmptyAddr(a.LocalRange) {
-		a.LocalRange = ""
-	}
+	a.LocalListenPort = h.LocalListenPort
```
Comment (abhishek9686) [inline models/api_host.go:52]: what is the purpose of `LocalListenPort`?

Comment (mattkasun) [reply models/api_host.go:52]: Probably no longer required but out of scope for this PR

Comment (abhishek9686) [reply models/api_host.go:52]: it was actually removed in PR #1949 , probably you need to rebase your branch with develop

Comment (mattkasun) [reply models/api_host.go]: removed

**Comment by theguy951357** [comment]
has to wait on PR#1951 before can test

File: logic/hosts.go [hunk 2]
```diff
@@ -315,3 +316,38 @@ func GetRelatedHosts(hostID string) []models.Host {
...
+// CheckHostPort checks host endpoints to ensures that hosts on the same server
+// with the same endpoint have different listen ports
+// in the case of 64535 hosts or more with same endpoint, ports will not be changed
+func CheckHostPorts(h *models.Host) {
+	portsInUse := make(map[int]bool)
+	hosts, err := GetAllHosts()
+	if err != nil {
+		return
+	}
+	for _, host := range hosts {
```
Comment (0xdcarns) [inline logic/hosts.go:329]: only hosts with the same endpoint should be considered here

File: models/host.go [hunk 3]
```diff
@@ -24,9 +24,8 @@ type Host struct {
 	Interface        string           `json:"interface" yaml:"interface"`
 	Debug            bool             `json:"debug" yaml:"debug"`
 	ListenPort       int              `json:"listenport" yaml:"listenport"`
-	LocalAddress     net.IPNet        `json:"localaddress" yaml:"localaddress"`
-	LocalRange       net.IPNet        `json:"localrange" yaml:"localrange"`
 	PublicListenPort int              `json:"public_listen_port" yaml:"public_listen_port"`
+	LocalListenPort  int              `json:"locallistenport" yaml:"locallistenport"`
```
Comment (0xdcarns) [inline models/host.go:28]: LocalListenPort was removed in previous PR

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
pass

File: mq/handlers.go [hunk 4]
```diff
@@ -141,7 +141,7 @@ func UpdateHost(client mqtt.Client, msg mqtt.Message) {
 			logger.Log(1, "error unmarshaling payload ", err.Error())
 			return
 		}
-		logger.Log(0, "recieved host update for host: ", id)
+		logger.Log(0, fmt.Sprintf("recieved host update: %+v\n", hostUpdate))
```
Comment (afeiszli) [inline mq/handlers.go:144]: can we make this log a 1 or 2?

**Review by afeiszli — COMMENTED** [review]
other than that one comment looks good to me

**Pull request merged**